src/searcher/mod.rs

```diff
@@ -212,6 +214,38 @@ impl Searcher {
         self.search_file_maybe_path(query_context, Some(path), &file, write_to)
     }
 
+    pub fn search_path_callback<P, TError: SinkError>(
```

Comment:
These sort of parallel the existing `Searcher` methods, just taking a callback instead of a `Sink` more or less

Didn't really think about trying to DRY up too much